src/components/LocalePicker/index.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import Picker from '../Picker';
+import Text from '../Text';
+import compose from '../../libs/compose';
+import {setLocale} from '../../libs/actions/App';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Permissions from '../../libs/Permissions';
+
+const propTypes = {
+    /** Indicates which locale the user currently has selected */
+    preferredLocale: PropTypes.string,
+
+    /** Indicates size of a picker component and whether to render the label or not */
+    size: PropTypes.oneOf(['normal', 'small']),
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    preferredLocale: CONST.DEFAULT_LOCALE,
+    size: 'normal',
+    betas: [],
+};
+
+const LocalePicker = ({
+    preferredLocale, translate, betas, size,
+}) => {
+    if (!Permissions.canUseInternationalization(betas)) {
+        return null;
+    }
+
+    const localesToLanguages = {
+        default: {
+            value: 'en',
+            label: translate('preferencesPage.languages.english'),
+        },
+        es: {
+            value: 'es',
+            label: translate('preferencesPage.languages.spanish'),
```

Comment:
```suggestion
            label: translate('preferencesPage.languages.es'),
```
Same for the above

Reply:
Sorry, but I'm confused about this one. Do you want me to change the translation keys for languages to be their 2 letter code instead of full language name (spanish -> es, english -> en)? I thought your point about `es` and `en` was to pass them as first argument to the `translate` function.